packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.stories.jsx

```diff
@@ -96,6 +97,29 @@ const Template = ({ ...args }) => {
   );
 };
 
+const LargeTemplateWithOptionalComponent = ({ ...args }) => {
+  const [data, setData] = useState(() => generateData({ rows: 100000 }));
+  const columns = useMemo(() => columnData, []);
+
+  const buildComponent = (index) => (
+    <OverflowMenu style={{ width: '28px' }}>
+      <OverflowMenuItem itemText={`Test item ${index}`} />
+    </OverflowMenu>
+  );
+
+  return (
+    <DataSpreadsheet
+      columns={columns}
+      data={data}
+      onDataUpdate={setData}
+      rowNumberingComponentPosition="Left"
+      rowNumberingComponent={buildComponent}
```

Comment:
i think `rowNumberingComponent` is a tad bit confusing. i think that because the documentation specifies usage of overflow menu that maybe we do an implementation where the API is called `rowActions` which accepts an array of actions that can be mapped over internally.

Reply:
So does the documentation specify it has to be overflow menu? I have it set so user provides custom component so it could be icon, button, or overflow menu.

Reply:
<img width="1840" alt="Screenshot 2024-06-17 at 1 02 33 PM" src="https://github.com/carbon-design-system/ibm-products/assets/6370760/3692e7f2-0701-4537-95b4-7b3aa3697f0f">

it's a bit hard to find because it's nested under an accordion, but it seems like it's mostly targeted towards overflow usage.

actually looking at this more i think maybe this should be set in the `columns` API. something like

```
const columnData = [
  {
    Header: 'Row Index',
    headerActions: [{ label: 'delete', onClick: () => {} }, ...],
    accessor: (row, index) => index,
...
```

let me pull in @matthewgallo on this since he's much more familiar with this component and i'd like to get his take on this.

Reply:
Sure, thanks

Reply:
after talking it over with @matthewgallo we think that a new prop called `renderHeaderOverflow` would work for this. the only main change we'd like to see is that this overflow is also accessible in the column header as per the usage guidelines. the prop could also provide a callback that distinguishes between a row and column header in case a separate set of actions are required for rows and columns. does that make sense?

Reply:
I see what you mean, but for the column, this is already customizable for the user to do. See how I am currently doing it for my code 

<img width="748" alt="Screenshot 2024-06-17 at 11 40 48 AM" src="https://github.com/carbon-design-system/ibm-products/assets/10100397/282d1571-5bd5-4472-9916-e8c9620b2281">

And then I provide my overflow menu item in here

<img width="1064" alt="Screenshot 2024-06-17 at 11 41 09 AM" src="https://github.com/carbon-design-system/ibm-products/assets/10100397/63336b16-80d6-4248-b479-a4d08f41d496">

Reply:
Result
![Screenshot 2024-06-17 at 11 44 27 AM](https://github.com/carbon-design-system/ibm-products/assets/10100397/65ec012a-528d-484a-8fac-e42a2719dc57)

Reply:
Hi @davidmenendez I discussed with Matt on this with some back and forth to get understanding. 

Currently, with column headers user can already provide custom component to it as with the example I provided above with how we have this in our code right now. 

For row headers on the row indexing numbering column there is no user control on this as it is hardcoded in. I explained to Matt how it is easier for Developer to provide their own custom component such as the overflow menu and if that menu has action events, it is easier for us to handle these action events rather than having to try and get them sorted through carbon products.

The changes I will be making is renaming the components to something more understandable like `renderRowHeader` and `renderRowHeaderDirection`. I'll also double check with the carbon guidelines, but storybook does have it's own section showing user how to utilize this feature and example is going to be using the overflowmenu. 

In future, if this is wanting to be changed around then that is fine but for now since column headers is already customizable, rather than making this not customizable and forcing it to be a pure string and having separate way of adding component. It is easier for now to leave it as is to avoid regressions. 

Any thoughts on this? 

I will update the PR in a bit with renaming the variables. Thanks

Reply:
this sounds good to me. if @matthewgallo has any questions or concerns i'll leave it up to him to post them here.

Reply:
Sounds like a plan for now, this way we don't interfere with any existing functionality, column headers already being able to be customized. 👍

Reply:
Thanks everyone, will leave a comment once I have the PR updated.